### PR TITLE
ENH: Archiver Timeplot address setters

### PR DESCRIPTION
**Pull request by zdomke**
Made changes to the address setters for ArchivePlotCurveItem and TimePlotCurveItem classes:

1. Archive curve object now has an address setter that replaces the function setArchiveChannel. The setter calls the TimePlotCurveItem address setter, then sets the new address as the archive channel. This results in the live and archive channels always being tied to the same address.
2. Both address setters now also check that the new address is different from the current address. If they're the same, the function returns.
3. Changed the way the ArchivePlotCurveItem address setter replaces the address's protocol. Now the pydm utility is used.
4. The TimePlotCurveItem address setter clears all live data tied to the previous channel. Similarly, the ArchivePlotCurveItem removes all archived data for the previous channel.

### CODE REVIEW

File: pydm/widgets/archiver_time_plot.py [hunk 1]
```diff
@@ -65,20 +65,28 @@ def to_dict(self) -> OrderedDict:
         dic_.update(super(ArchivePlotCurveItem, self).to_dict())
         return dic_
 
-    def setArchiveChannel(self, address: str) -> None:
+    @TimePlotCurveItem.address.setter
```
Comment (nstelter-slac) [inline pydm/widgets/archiver_time_plot.py:68]: im wondering why '@TimePlotCurveItem.address.setter' instead of a '@address.setter' for the current class?

Comment (zdomke) [reply pydm/widgets/archiver_time_plot.py:68]: This was a way for me to reference the super class' property without overriding the getter.

Looking back through the code, this only saves a few lines of code while being harder to read. Instead we could consider this if you think it is easier to read:
```suggestion
    @property
    def address(self):
        return super().address
        
    @address.setter
    def address(self, new_address: str) -> None:
```

Comment (nstelter-slac) [reply pydm/widgets/archiver_time_plot.py:68]: i like this, find it easier to read

**Review by jbellister-slac — APPROVED** [review]
Looks good, thanks!

**Pull request merged**